src/app_service/service/community/service.nim

```diff
@@ -371,10 +373,14 @@ QtObject:
       if community.settings.id == "":
         community.settings = prev_community.settings
 
+      var deletedCategories: seq[string] = @[]
+
       # category was added
       if(community.categories.len > prev_community.categories.len):
         for category in community.categories:
           if findIndexById(category.id, prev_community.categories) == -1:
+            self.allCommunities[community.id].categories.add(category)
+            self.joinedCommunities[community.id].categories.add(category)
```

Comment:
We also need to ensure newly added categories are stored in our service, otherwise chats will not make it into their categories when new categories are created including selected chats.